src/hooks/useReadyWithDimensions.ts

```diff
@@ -0,0 +1,31 @@
+import {useEffect, useState} from 'react';
+import {Dimensions} from 'react-native';
+import CONST from '@src/CONST';
+
+const useReadyWithDimensions = (isEnabled = true) => {
+    const [isReady, setIsReady] = useState(false);
+
+    useEffect(() => {
+        if (!isEnabled) {
+            return;
+        }
+        const timer = setTimeout(() => {
```

Comment:
[Code patterns to avoid](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#L56)

Could you please explain the necessity of using `setTimeout` in this case? We should clarify the issue to ensure that the solution addresses it.